src/main/kotlin/io/libp2p/core/Libp2pException.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2019 BLK Technologies Limited (web3labs.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.libp2p.core
+
+class Libp2pException : RuntimeException {
```

Comment:
This might be too generic an exception?

Reply:
Just took it from Web3 code. We may use it as superclass for all libp2p exceptions